rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/dateFilter.spec.ts

```diff
@@ -0,0 +1,179 @@
+      const wrapper = mountDateFilter();
+      const checkbox = wrapper.find('input[type="checkbox"]');
+      expect((checkbox.element as HTMLInputElement).checked).toBe(false);
+    });
+  });
+  describe("Rendering", () => {
+    it("renders checkbox and dropdown based on enabled property", async () => {
+      const wrapper = mountDateFilter({
+        props: {
+          modelValue: {
+            enabled: true,
+            datetime: "2022-01-02T00:00:00",
+          },
+        },
+      });
+      await wrapper.vm.$nextTick();
+      const checkbox = wrapper.find('input[type="checkbox"]');
+      expect(checkbox.exists()).toBe(true);
+      const dropdown = wrapper.find(".dropdown");
+      expect(dropdown.exists()).toBe(true);
+    });
+    it("renders provided slot content in label", async () => {
+      const slotContent = "Provided slot content";
+      const wrapper = mountDateFilter({ slots: { default: slotContent } });
+      await wrapper.vm.$nextTick();
+      const label = wrapper.find("label");
+      expect(label.text()).toBe(slotContent);
+    });
+    it("binds datetime to text input", async () => {
+      const wrapper = mountDateFilter({
+        props: {
+          modelValue: {
+            enabled: true,
+            datetime: "2022-01-01T00:00:00",
+          },
+        },
+      });
+      await wrapper.vm.$nextTick();
+      const dateTimePicker = wrapper.findComponent({ name: "DateTimePicker" });
+      expect(dateTimePicker.exists()).toBe(true);
+      expect(dateTimePicker.props("modelValue")).toBe("2022-01-01T00:00:00");
+    });
+  });
+  describe("User Interactions", () => {
+    it("toggles enabled state when checkbox is clicked", async () => {
+      const wrapper = mountDateFilter();
+      const checkbox = wrapper.find('input[type="checkbox"]');
+      await checkbox.setValue(true);
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.enabled).toBe(true);
+      await checkbox.setValue(false);
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.enabled).toBe(false);
+    });
+    it("updates datetime when DateTimePicker emits update:modelValue", async () => {
+      const wrapper = mountDateFilter({
+        props: {
+          modelValue: { enabled: true },
+        },
+      });
+      await wrapper.vm.$nextTick();
+      const dateTimePicker = wrapper.findComponent(DateTimePicker);
+      const newDate = "2022-01-02T00:00:00";
+      dateTimePicker.vm.$emit("update:modelValue", newDate);
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.datetime).toBe(newDate);
+    });
+    it("emits update:modelValue event when enabled changes", async () => {
+      const wrapper = mountDateFilter();
+      const checkbox = wrapper.find('input[type="checkbox"]');
+      await checkbox.setValue(true);
+      await wrapper.vm.$nextTick();
+      const emitted = wrapper.emitted("update:modelValue");
+      expect(emitted).toBeTruthy();
+      if (emitted) {
```

Comment:
please remove this if and just leave the line expect(emitted[0]).toEqual([{ enabled: true, datetime: "" }]);